server/auth/local.js

```diff
@@ -0,0 +1,22 @@
+const { User } = require('../db').models;
+const LocalStrategy = require('passport-local').Strategy;
+const options = { usernameField: 'email' };
+
+function checkPassword(user, password, done) {
+  return user.validPassword(password)
+    .then(valid => {
+      if (!valid) return done(null, false, { message: 'Incorrect password.' });
+      return done(null, user);
+    })
+    .catch(err => done(err));
+}
+
+function authenticate(email, password, done) {
+  return User.find({ where: { email } })
+    .then(user => {
+      if (!user) return done(null, false, { message: 'Incorrect username.' });
```

Comment:
We are not using a username, but an email address.

Reply:
Also, it's not the best idea security-wise to specify what isn't correct, email or password since it gives more info than it should. Change both of these messages to something like 'Incorrect email or password.', extract it to separate variable and use in both, `checkPassword` and `authenticate`.